wiring/src/spark_wiring_string.cpp

```diff
@@ -217,7 +217,8 @@ String & String::copy(const char *cstr, unsigned int length)
 		return *this;
 	}
 	len = length;
-	strcpy(buffer, cstr);
+	memcpy(buffer, cstr, length);
+	buffer[len] = 0;
```

Comment:
Might it not be clearer if it was `buffer[len] = '\0';`? My C/C++ knowledge is woefully behind y'all's so sorry if the suggestion is silly.